jgrapht-core/src/main/java/org/jgrapht/alg/connectivity/BlockCutpointGraph.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * (C) Copyright 2007-2017, by France Telecom and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.connectivity;
+
+import java.util.*;
+
+import org.jgrapht.*;
+import org.jgrapht.graph.*;
+
+/**
+ * A Block-Cutpoint graph (also known as a block-cut tree).
+ * If $G$ is a graph, the block-cutpoint graph of $G$, denoted $BC(G)$ is the simple bipartite graph with
+ * bipartition $(A, B)$ where $A$ is the set of <a href="http://mathworld.wolfram.com/ArticulationVertex.html">cut-vertices</a> (also known as articulation points) of $G$,
+ * and $B$ is the set of <a href="http://mathworld.wolfram.com/Block.html">blocks</a> of $G$. $BC(G)$ contains an edge $(a,b)$ for $a \in A$ and $b \in B$ if and only if
+ * block $b$ contains the cut-vertex $a$.
+ * A vertex in $G$ is a cut-vertex if removal of the vertex from $G$ (and all edges incident to this vertex) increases
+ * the number of connected components in the graph. A block of $G$ is a maximal connected subgraph $H \subseteq G$ so
+ * that $H$ does not have a cut-vertex. Note that if $H$ is a block, then either $H$ is 2-connected, or $|V(H)| \leq 2$.
+ * Each pair of blocks of $G$ share at most one vertex, and that vertex is a cut-point in $G$. $BC(G)$ is a tree in which
+ * each leaf node corresponds to a block of $G$.
+ * <p>
+ * Note: the block-cutpoint graph is not changed when the underlying graph is changed.
+ *
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ *
+ * @author France Telecom S.A
+ * @author Joris Kinable
+ * @since July 5, 2007
+ */
+public class BlockCutpointGraph<V, E>
+        extends SimpleGraph<Graph<V, E>, DefaultEdge>
```

Comment:
In the future we should create a static method which creates this graph given an input graph instead of having it extend `SimpleGraph` and perform the construction in the constructor.